service/storage_service.cc

```diff
@@ -534,12 +531,14 @@ future<storage_service::nodes_to_notify_after_sync> storage_service::sync_raft_t
 
```

Comment:
> when multiple nodes are removed in succession

The problem seems broader. To hit it, it's enough to add any topology change to group 0 after a node is removed.